elfloader-tool/src/drivers/driver.c

```diff
@@ -53,3 +53,35 @@ int initialise_devices(void)
 
     return 0;
 }
+
+static int init_device_non_boot(struct elfloader_device *dev)
+{
+    struct elfloader_driver **drvp = __start__driver_list;
+
+    while (drvp < __stop__driver_list) {
+        struct elfloader_driver *drv = *drvp;
+        int ret = table_has_match(dev->compat, drv->match_table);
+        if (ret >= 0 && drv->init_non_boot) {
+            dev->drv = drv;
+            drv->init_non_boot(dev, drv->match_table[ret].match_data);
+        }
+        drvp++;
+    }
+
+    return 0;
+
+}
+
+
+int initialise_devices_non_boot(void)
+{
+    for (unsigned int i = 0; i < ARRAY_SIZE(elfloader_devices); i++) {
+        int ret = init_device_non_boot(&elfloader_devices[i]);
+        if (ret) {
+            return ret;
+        }
+    }
+
+    return 0;
+
```

Comment:
I'd remove the empty lines around `return`, it just looks weird.

But again, if no one is checking the return value, just make this function void and keep going.